plenoptic/synthesize/synthesis.py

```diff
@@ -1075,117 +972,89 @@ def save(self, file_path, save_model_reduced=False, attrs=['model'],
-        appropriate location:
-
-        >>> model = po.simul.PooledRGC(1)
-        >>> metamer = po.synth.Metamer(img, model)
-        >>> metamer.synthesize(max_iter=10, store_progress=True)
-        >>> metamer.save('metamers.pt', save_model_reduced=True)
-        >>> metamer_copy = po.synth.Metamer.load('metamers.pt',
-                                                 model_constructor=po.simul.PooledRGC.from_state_dict_reduced,
-                                                 cache_dir="/home/user/Desktop/metamers/windows_cache")
+        Note that you must create a new instance of the Synthesis object and
+        *then* load.
 
         """
         tmp_dict = torch.load(file_path, map_location=map_location, pickle_module=dill)
-        device = torch.device(map_location)
-        if not isinstance(model_attr_name, list):
-            model_attr_name = [model_attr_name]
-        if not isinstance(model_constructor, list):
-            model_constructor = [model_constructor]
-        base_signal = tmp_dict.pop('base_signal').to(device)
-        models = {}
-        for attr, constructor in zip(model_attr_name, model_constructor):
-            model = tmp_dict.pop(attr)
-            if isinstance(model, dict):
-                for k, v in state_dict_kwargs.items():
-                    warnings.warn("Replacing state_dict key %s, value %s with kwarg value %s" %
-                                  (k, model.pop(k, None), v))
-                    model[k] = v
-                # then we've got a state_dict_reduced and we need the model_constructor
-                model = constructor(model)
-                # want to make sure the dtypes match up as well
-                model = model.to(device, base_signal.dtype)
-            models[attr] = model
-        loss_function = tmp_dict.pop('loss_function', None)
-        loss_function_kwargs = tmp_dict.pop('loss_function_kwargs', {})
-        synth = cls(base_signal, loss_function=loss_function,
-                    loss_function_kwargs=loss_function_kwargs, **models)
+        for k in check_attributes:
+            if not hasattr(self, k):
+                raise Exception("All values of `check_attributes` should be attributes set at"
+                                f" initialization, but got attr {k}!")
+            if isinstance(getattr(self, k), torch.Tensor):
+                try:
+                    if not torch.allclose(getattr(self, k), tmp_dict[k]):
+                        raise Exception(f"Saved and initialized {k} are different! Initialized: {getattr(self, k)}"
+                                        f", Saved: {tmp_dict[k]}, difference: {getattr(self, k) - tmp_dict[k]}")
+                except RuntimeError:
+                    raise Exception(f"Attribute {k} is a different shape in saved and initialized versions!"
+                                    f" Initialized: {getattr(self, k).shape}, Saved: {tmp_dict[k].shape}")
+            elif k == 'loss_function':
```

Comment:
Confused as to what's going on here.

Reply:
I added some comments trying to explain this